controller/backing_image_controller.go

```diff
@@ -310,22 +345,530 @@ func (bic *BackingImageController) syncBackingImage(key string) (err error) {
+		bi.Status.V2FirstCopyStatus = longhorn.BackingImageStateFailed
+		return bic.cleanupFirstFailedV2Copy(bi)
+	}
+
+	// If the first v2 copy is in progress, we wait for the next reconciliation
+	isInProgress := bic.isFirstV2CopyInState(bi, longhorn.BackingImageStateInProgress)
+	if isInProgress {
+		bi.Status.V2FirstCopyStatus = longhorn.BackingImageStateInProgress
+		bi.Spec.DiskFileSpecMap[bi.Status.V2FirstCopyDisk] = &longhorn.BackingImageDiskFileSpec{DataEngine: longhorn.DataEngineTypeV2}
+		return nil
+	}
+
+	// Wait for the first v1 file to be ready.
+	bids, err := bic.ds.GetBackingImageDataSource(bi.Name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Warn("Backing image data source not found when preparing first v2 copy, reconcile later")
+			return nil
+		}
+		return errors.Wrap(err, "failed to get the backing image data source when preparing first v2 copy")
+	}
+	if !bids.Spec.FileTransferred {
+		log.Info("Backing image data source has not prepared the first v1 file, reconcile later")
+		return nil
+	}
+	firstV1FileDiskUUID := bids.Spec.DiskUUID
+	if status, exists := bi.Status.DiskFileStatusMap[firstV1FileDiskUUID]; exists && status.State != longhorn.BackingImageStateReady {
+		log.Infof("The first v1 file copy is not ready, reconcile later")
+		return nil
+	}
+
+	firstV2CopyNode, firstV2CopyDiskName, err := bic.findReadyNodeAndDiskForFirstV2Copy(bi, firstV1FileDiskUUID, log)
+	if err != nil {
+		return errors.Wrap(err, "failed to find a ready disk for the first v2 backing image copy")
+	}
+
+	firstV2CopyNodeName := firstV2CopyNode.Name
+	firstV2CopyDiskUUID := firstV2CopyNode.Status.DiskStatus[firstV2CopyDiskName].DiskUUID
+	log.Infof("Founded the ready node %v disk %v for the first v2 backing image copy", firstV2CopyNodeName, firstV2CopyDiskUUID)
+
+	if bic.v2CopyBackoff.IsInBackOffSinceUpdate(firstV2CopyDiskUUID, time.Now()) {
+		log.Debugf("Skip preparing first v2 backing image copy to disk %v since it is still in the backoff window", firstV2CopyDiskUUID)
+		return nil
+	}
+
+	// Get proxy client for creating v2 backing image copy
+	engineClientProxy, err := bic.getEngineClientProxy(firstV2CopyNodeName, longhorn.DataEngineTypeV2, log)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get the engine client proxy for node %v", firstV2CopyNodeName)
+	}
+	defer engineClientProxy.Close()
+
+	// Create v2 backing image copy by downloading the data from first file copy
+	fileDownloadAddress, err := bic.getFileDownloadAddress(bi, firstV1FileDiskUUID)
+	if err != nil {
+		return errors.Wrap(err, "failed to get the v1 file download address when preparing first v2 copy")
+	}
+	_, err = engineClientProxy.SPDKBackingImageCreate(bi.Name, bi.Status.UUID, firstV2CopyDiskUUID, bi.Status.Checksum, fileDownloadAddress, "", uint64(bi.Status.Size))
+	if err != nil {
+		if types.ErrorAlreadyExists(err) {
+			log.Infof("backing image already exists when preparing first v2 copy on disk %v", firstV2CopyDiskUUID)
+		}
```

Comment:
Should we blindly overwrite the existing file?

Reply:
No, it could be in the middle of creating the lvol
It will complicated the case.